k2/python/k2/rnnt_loss.py

```diff
@@ -531,9 +657,23 @@ def get_rnnt_prune_ranges(
     assert T1 in [T, T + 1]
     assert py_grad.shape == (B, S + 1, T)
     assert boundary.shape == (B, 4)
-    assert s_range >= 1
+    assert S >= 1
+    assert T >= S
+
+    # s_range > S means we won't prune out any symbols. To make indexing with
+    # ranges runs normally, s_range should be equal to or less than ``S + 1``.
     if s_range > S:
-        s_range = S
+        s_range = S + 1
```

Comment:
The buggy code that raises nan or inf loss is here.  It only occurs when `S==1` and `modified==False`, because when `modified==False`  the transitions can only go up and right, so `s_range` MUST be equal to or greater than 2, or no path can survive pruning.